src/v/kafka/server/group_manager.cc

```diff
@@ -103,7 +103,14 @@ ss::future<> group_manager::stop() {
         e.second->as.request_abort();
     }
 
-    return _gate.close();
+    return _gate.close().then([this] {
+        /**
+         * cancel all pending group opeartions
+         */
+        for (auto& [_, group] : _groups) {
+            group->shutdown();
```

Comment:
I'm surprised that the group shutdown doesn't happen before the gate is closed. any specific reason for this ordering?

Reply:
i made it like this to make sure no operations will change the group state after it is shutdown.